tests/apps/workflowsapp/app.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/gorilla/mux"
+
+	"github.com/dapr/components-contrib/workflows"
+	"github.com/dapr/dapr/tests/apps/utils"
+
+	"go.temporal.io/api/workflowservice/v1"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/worker"
+	"go.temporal.io/sdk/workflow"
+)
+
+var appPort = 3000
+
+func init() {
+	p := os.Getenv("PORT")
+	if p != "" && p != "0" {
+		appPort, _ = strconv.Atoi(p)
+	}
+}
+
+type testCommandRequest struct {
+	Message string `json:"message,omitempty"`
+}
+
+type appResponse struct {
+	Message   string `json:"message,omitempty"`
+	StartTime int    `json:"start_time,omitempty"`
+	EndTime   int    `json:"end_time,omitempty"`
+}
+
+var httpClient = utils.NewHTTPClient()
+
+// indexHandler is the handler for root path
+func indexHandler(w http.ResponseWriter, r *http.Request) {
+	log.Println("indexHandler is called")
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(appResponse{Message: "OK"})
+}
+
+// testHandler is the handler for end-to-end test entry point
+// test driver code call this endpoint to trigger the test
+func testHandler(w http.ResponseWriter, r *http.Request) {
+	// Retrieve request body contents
+	var commandBody testCommandRequest
+	err := json.NewDecoder(r.Body).Decode(&commandBody)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		json.NewEncoder(w).Encode(appResponse{
+			Message: err.Error(),
+		})
+		return
+	}
+
+	// Trigger the test
+	startTime := epoch()
+	statusCode, res := startTest(commandBody)
+	res.StartTime = startTime
+	res.EndTime = epoch()
+
+	w.WriteHeader(statusCode)
+	json.NewEncoder(w).Encode(res)
+}
+
+func startTest(commandRequest testCommandRequest) (int, appResponse) {
```

Comment:
Two things on this:

1. Should also test the "workflow terminate" method
2. Tests should work with both HTTP and gRPC (see other apps how they do it)